### PR TITLE
chore: add instructions for windows and fix failing unit tests

### PR DESCRIPTION
**Pull request by RealAnna**
## This PR

- fixes unit tests that failed on windows machines
- clarifies usage of flagd on a windows machine

### Related Issues

https://github.com/open-feature/flagd/issues/630

### Notes

The learning from this pr were:

1.  Always prefer using the go test suite tempdir instead of relying on os library to allocate and delete test folders, this simplifies code and avoids os-specific issues
2.  Make sure to always close file pointers, since leaving dangling ones tends to cause access errors in Windows aka error file already open. 
3.  When running via docker one must make sure that the file is in the same filesystem where the docker container runs than the watcher will work. AKA on Windows it is good as long as everything comes from WSL filesystem

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -98,6 +98,13 @@ Experiment with flagd in your browser using [the Killercoda tutorial](https://ki
       -d '{"flagKey":"myStringFlag","context":{}}' -H "Content-Type: application/json"
     ```
 
+    For Windows use bash or use the following:
```
Comment (thisthat) [inline README.md:101]:
```suggestion
    For Windows we recommend using a [WSL](https://learn.microsoft.com/en-us/windows/wsl/install) terminal. 
    Otherwise, use the following with `cmd` or `powershell` 
```

Comment (RealAnna) [reply README.md:101]: powershell wont work with this curl, but I will commit the other suggestions 👍

File: README.md [hunk 2]
```diff
@@ -49,6 +49,7 @@ Experiment with flagd in your browser using [the Killercoda tutorial](https://ki
     ```
 
     Or use docker:
+   (for Windows use WSL both for the file path and the docker commands. This flagd on docker command WILL NOT WORK on Windows due to the mixed filesystems [#8479](https://github.com/docker/for-win/issues/8479)!! See also [this medium article](https://levelup.gitconnected.com/docker-desktop-on-wsl2-the-problem-with-mixing-file-systems-a8b5dcd79b22) )
```
Comment (Kavindu-Dodan) [inline README.md:52]: I would reword this section to something like the below (italic text & in a new line), 

```
    _Note - In Windows, use WSL system for both the file location and Docker runtime. Mixed file systems does not 
   work and this is a limitation of Docker (https://github.com/docker/for-win/issues/8479)_
```

I have mixed feelings on linking to the medium article as this is a premium access only. Plus, rewording to highlight this comes from Docker rather than flagd :)

File: README.md [hunk 3]
```diff
@@ -73,7 +74,7 @@ Experiment with flagd in your browser using [the Killercoda tutorial](https://ki
       --uri file:./example_flags.flagd.json
     ```
 
-    Or use docker:
+    Or use docker (!! this does not work on Windows without using WSL !!):
```
Comment (Kavindu-Dodan) [inline README.md:77]: Moving this to italic text & in a new line with a `Note`

And rather than saying `does not work`, reword to say something like "In Windows, this requires WSL"

File: README.md (hunk 2 above)
Comment (toddbaert) [reply README.md:52]: agreed.

**Review by toddbaert — COMMENTED** [review]
LGTM but I agree with @Kavindu-Dodan 's suggestions.

File: README.md [hunk 4]
```diff
@@ -50,7 +50,7 @@ Experiment with flagd in your browser using [the Killercoda tutorial](https://ki
 
     Or use docker:
     _Note - In Windows, use WSL system for both the file location and Docker runtime. Mixed file systems does not
-    work and this is a limitation of Docker (https://github.com/docker/for-win/issues/8479)_
+    work and this is a [limitation of Docker] (https://github.com/docker/for-win/issues/8479)_
```
Comment (thisthat) [inline README.md:53]:
```suggestion
    work and this is a [limitation of Docker](https://github.com/docker/for-win/issues/8479)_
```

**Review by Kavindu-Dodan — APPROVED** [review]
lgtm
